dali/kernels/imgproc/color_manipulation/color_space_conversion_kernel.cuh

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) 2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_COLOR_SPACE_CONVERSION_KERNEL_CUH_
+#define DALI_KERNELS_IMGPROC_COLOR_MANIPULATION_COLOR_SPACE_CONVERSION_KERNEL_CUH_
+
+#include <cuda_runtime_api.h>
+#include "dali/kernels/imgproc/color_manipulation/color_space_conversion_impl.h"
+
+namespace dali {
+namespace kernels {
+namespace color {
+
+template <typename Out, typename In>
+DALI_HOST_DEV struct RGB_to_BGR_Converter {
```

Comment:
If Out==In would it compile out `ConvertSatNorm` ?

Reply:
It's optimized out, AFAIK